hermes/bin/src/runtime_extensions/hermes/doc_sync/host.rs

```diff
@@ -60,25 +65,89 @@ impl HostSyncChannel for HermesRuntimeContext {
         _self_: Resource<SyncChannel>,
         _name: ChannelName,
     ) -> wasmtime::Result<Result<bool, Errno>> {
-        todo!()
+        Ok(Ok(true))
     }
 
-    /// Post the document to a channel
-    ///  
-    /// **Parameters**
-    ///
-    /// None
+    /// Post a document to IPFS and broadcast via `PubSub`.
     ///
-    /// **Returns**
-    ///
-    /// - `ok(true)`: Channel Closed and resources released.
-    /// - `error(<something>)`: If it gets an error closing.
+    /// Executes the 4-step workflow:
+    /// 1. Add to IPFS (`file_add`)
+    /// 2. Pin (`file_pin`)
+    /// 3. Pre-publish (TODO #630)
+    /// 4. Publish to `PubSub` (`pubsub_publish`)
     fn post(
         &mut self,
         _self_: Resource<SyncChannel>,
-        _doc: DocData,
+        doc: DocData,
     ) -> wasmtime::Result<Result<DocLoc, Errno>> {
-        todo!()
+        tracing::info!("📤 Posting {} bytes to doc-sync channel", doc.len());
+
+        // Step 1: Add document to IPFS
+        let ipfs_path = match self.file_add(doc.clone())? {
```

Comment:
As I mention before, the `file_add` -> `hermes_ipfs_add_file` has this `    ipfs.apps.pinned_file(app_name.clone(), &ipfs_path)?;`

Reply:
sorry, I see now. I removed the explicit pin.